perceval/backends/core/weblate.py

```diff
@@ -0,0 +1,374 @@
+
+        self.api_token = api_token
+        self.project = project
+        self.max_retries = max_retries
+        self.sleep_time = sleep_time
+        self.sleep_for_rate = sleep_for_rate
+        self.min_rate_to_sleep = min_rate_to_sleep
+
+        self.client = None
+
+    def search_fields(self, item):
+        """Add search fields to an item.
+
+        It adds the values of `metadata_id` plus the `project`.
+
+        :param item: the item to extract the search fields values
+
+        :returns: a dict of search fields
+        """
+        search_fields = {
+            DEFAULT_SEARCH_FIELD: self.metadata_id(item)
+        }
+
+        return search_fields
+
+    def fetch(self, category=CATEGORY_CHANGES, from_date=DEFAULT_DATETIME):
+        """Fetch changes from Weblate API.
+
+        The method retrieves the changes that occurred on a project via the Weblate API.
+
+        :param category: the category of items to fetch
+
+        :returns: a generator of data
+        """
+        if not from_date:
+            from_date = DEFAULT_DATETIME
+
+        kwargs = {
+            'from_date': from_date
+        }
+        items = super().fetch(category, **kwargs)
+
+        return items
+
+    def fetch_items(self, category, **kwargs):
+        """Fetch Change items
+
+        :param category: the category of items to fetch
+        :param kwargs: backend arguments
+
+        :returns: a generator of items
+        """
+        logger.info("Fetching data")
+        from_date = kwargs['from_date']
+
+        changes_groups = self.client.changes(from_date=from_date)
+        for changes in changes_groups:
+            for change in changes:
+                user = change.get('user', None)
+                if user:
+                    change['user_data'] = self.client.user(user)
+                author = change.get('author', None)
+                if author:
+                    change['author_data'] = self.client.user(user)
+
+                yield change
+
+        logger.info("Fetch process completed")
+
+    @classmethod
+    def has_archiving(cls):
+        """Returns whether it supports archiving items on the fetch process.
+
+        :returns: this backend supports items archive
+        """
+        return True
+
+    @classmethod
+    def has_resuming(cls):
+        """Returns whether it supports to resume the fetch process.
+
+        :returns: this backend supports items resuming
+        """
+        return True
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from a Change item."""
+
+        return str(item['id'])
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from a Change item.
+
+        The timestamp is based on the current time when the hit was extracted.
+        This field is not part of the data provided by Google API. It is added
+        by this backend.
```

Comment:
This message is completely wrong and references `googlehits`. Please make reference about how the date is obtained here.